File: Problems/Arrays/Contains-Duplicate/Contains-Duplicate.cs
using System;
using System.Collections.Generic;

// Using hash table - Time O(n)

public class Solution
{
    public bool ContainsDuplicate(int[] nums)
    {
        HashSet<int> set = new HashSet<int>(nums);

        return set.Count != nums.Length;
    }
}
class Program
{
    static void Main(string[] args)
    {
        int[] nums = {1, 2, 3, 1};

        // Print input
        Console.Write("Input: nums = [");
        for (int i = 0; i < nums.Length; i++)
        {
            Console.Write(nums[i] + "");
            if (i < nums.Length - 1)
            {
                Console.Write(", ");
            }
        }
        Console.WriteLine("]");

        Solution sol = new Solution();
        bool result = sol.ContainsDuplicate(nums);

        // Print output
        Console.WriteLine("Output: " + result);
    }
}

File: Problems/Arrays/Contains-Duplicate/Contains-Duplicate.cpp
#include <unordered_set>
#include <algorithm>
#include <iostream>
#include <vector>

// Using hash table - Time O(n)

class Solution {
public:
    bool containsDuplicate(std::vector<int> &nums)
    {
        std::unordered_set<int> s(begin(nums), end(nums));
        return s.size() != nums.size();
    }
};

int main()
{
    std::vector<int> nums = {1, 2, 3, 1};

    // Print input
    std::cout << "Input: nums = [";
    for (const auto &num : nums)
    {
        std::cout << num << "";
        if (&num != &nums.back())
        {
            std::cout << ", ";
        }
    }
    std::cout << "]" << std::endl;

    Solution sol;
    bool result = sol.containsDuplicate(nums);

    // Print output
    std::cout << "Output: " << std::boolalpha << result << std::endl;

    return 0;
}